boards.txt

```diff
@@ -1553,6 +1553,93 @@ m5stack-core-esp32.menu.DebugLevel.verbose.build.code_debug=5
 
 ##############################################################
 
+m5stack-fire.name=M5Stack-FIRE
+
+m5stack-fire.upload.tool=esptool
+m5stack-fire.upload.maximum_size=1310720
+m5stack-fire.upload.maximum_data_size=327680
+m5stack-fire.upload.wait_for_upload_port=true
+
+m5stack-fire.serial.disableDTR=true
+m5stack-fire.serial.disableRTS=true
+
+m5stack-fire.build.mcu=esp32
+m5stack-fire.build.core=esp32
+m5stack-fire.build.variant=m5stack_fire
+m5stack-fire.build.board=M5STACK_FIRE
+
+m5stack-fire.build.f_cpu=240000000L
+m5stack-fire.build.flash_size=16MB
+m5stack-fire.build.flash_freq=40m
+m5stack-fire.build.flash_mode=dio
+m5stack-fire.build.boot=dio
+m5stack-fire.build.partitions=default
+m5stack-fire.build.defines=
+
+m5stack-fire.menu.PSRAM.disabled=Disabled
+m5stack-fire.menu.PSRAM.disabled.build.defines=
+m5stack-fire.menu.PSRAM.enabled=Enabled
+m5stack-fire.menu.PSRAM.enabled.build.defines=-DBOARD_HAS_PSRAM -mfix-esp32-psram-cache-issue
+
+m5stack-fire.menu.PartitionScheme.default=Default
+m5stack-fire.menu.PartitionScheme.default.build.partitions=default
+m5stack-fire.menu.PartitionScheme.minimal=Minimal (2MB FLASH)
+m5stack-fire.menu.PartitionScheme.minimal.build.partitions=minimal
+m5stack-fire.menu.PartitionScheme.no_ota=No OTA (Large APP)
+m5stack-fire.menu.PartitionScheme.no_ota.build.partitions=no_ota
+m5stack-fire.menu.PartitionScheme.no_ota.upload.maximum_size=2097152
+m5stack-fire.menu.PartitionScheme.min_spiffs=Minimal SPIFFS (Large APPS with OTA)
+m5stack-fire.menu.PartitionScheme.min_spiffs.build.partitions=min_spiffs
+m5stack-fire.menu.PartitionScheme.min_spiffs.upload.maximum_size=1966080
```

Comment:
those partitions make no sense. let's discuss this and add new options :)

Reply:
Yes, there is no partition for 16MB Flash, should add new options.